pkg/cmd/server/kubernetes/node/options/options.go

```diff
@@ -0,0 +1,211 @@
+package node
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+	kubeproxyoptions "k8s.io/kubernetes/cmd/kube-proxy/app"
+	kubeletoptions "k8s.io/kubernetes/cmd/kubelet/app/options"
+	"k8s.io/kubernetes/pkg/apis/componentconfig"
+	kubeletcni "k8s.io/kubernetes/pkg/kubelet/network/cni"
+	kubelettypes "k8s.io/kubernetes/pkg/kubelet/types"
+
+	configapi "github.com/openshift/origin/pkg/cmd/server/api"
+	cmdutil "github.com/openshift/origin/pkg/cmd/util"
+	cmdflags "github.com/openshift/origin/pkg/cmd/util/flags"
+	"github.com/openshift/origin/pkg/cmd/util/variable"
+	"github.com/openshift/origin/pkg/network"
+)
+
+// Build creates the core Kubernetes component configs for a given NodeConfig, or returns
+// an error
+func Build(options configapi.NodeConfig) (*kubeletoptions.KubeletServer, *componentconfig.KubeProxyConfiguration, error) {
```

Comment:
I'd prefer to build these separately.  Any reason that they *have* to be built together?

Reply:
Yeah, there is data calculated for kubelet server that is an input to proxy config.  I'd prefer not to separate that in this step (we can do that once kubelet config moves to beta).